### ISSUE TITLE
OBS signing key has expired (The gpg key signing file 'repomd.xml' has expired)

### ISSUE THREAD

**Issue by thomasmerz** [rear/rear#3289]
https://github.com/rear/rear/issues/2309 is back:

```
Warning: The gpg key signing file 'repomd.xml' has expired.
  Repository:       Relax-and-Recover snapshot packages (15.5)
  Key Fingerprint:  985D 26DB 1764 72E4 5FFE 5FF5 1B4D D1E2 FFC8 DD5F
  Key Name:         Archiving:Backup:Rear OBS Project <Archiving:Backup:Rear@build.opensuse.org>
  Key Algorithm:    RSA 2048
  Key Created:      Thu 17 Mar 2022 04:08:05 PM CET
  Key Expires:      Sat 25 May 2024 05:08:05 PM CEST (EXPIRED)
  Rpm Name:         gpg-pubkey-ffc8dd5f-62334ed5
```

* ReaR version ("/usr/sbin/rear -V"): Relax-and-Recover 2.7 / 2022-07-13

* If your ReaR version is not the current version, explain why you can't upgrade: see title of issue 😉 

* OS version ("cat /etc/os-release" or "lsb_release -a" or "cat /etc/rear/os.conf"): openSUSE Leap 15.5

**Comment by jsmeix** [comment]
@thomasmerz 
as always I have no idea how that OBS key stuff actually works.
To me all that is obscure and behaves inexplicably.

So I need details how I could reproduce your issue.
Ideally exact command lines that I could type in
and what their output is on your system
so I could compare your output with mine.
I do not use graphical GUI tools so I could not
reproduce issues with graphical GUI tools.

FYI
what I did so far
(basically blind attempts to "see" something):

https://build.opensuse.org/projects/Archiving:Backup:Rear/signing_keys
```
Expires on    2026-09-30 
```
which matches
```
# osc signkey Archiving:Backup:Rear | gpg
...
pub   rsa2048 2020-01-17 [SC] [expires: 2026-09-30]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F

# osc signkey Archiving:Backup:Rear:15.5 | gpg
...
pub   rsa2048 2020-01-17 [SC] [expires: 2026-09-30]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F
```

See also
https://lists.opensuse.org/archives/list/buildservice@lists.opensuse.org/message/PUY3UP37MA7EYS4XEWGDJRUYBHYNNIRR/
so I tried that
```
# osc signkey --extend Archiving:Backup:Rear
<status code="ok" />
```

But contradictingly
```
# curl https://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/repodata/repomd.xml.key | gpg
...
pub   rsa2048 2020-01-17 [SC] [expired: 2024-05-25]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F
```
but this somehow matches what
http://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/repodata/
shows
```
Name                      Last Modified
...
repomd.xml.key            10/19/2023
```

But on June 03 2024 I had already triggered a rebuild
of Archiving:Backup:Rear rear-2.7 for all build targets:
https://github.com/rear/rear/issues/3235#issuecomment-2144590757

???

**Comment by jsmeix** [comment]
I tried a rebuild:
```
# osc rebuild Archiving:Backup:Rear rear-2.7 15.5
ok

# osc results -v Archiving:Backup:Rear rear-2.7 | grep '^15.5'
15.5  x86_64   rear-2.7  building: building on i04-ch1d:2
15.5  ppc64le  rear-2.7  building: building on obs-power9-15:10

# osc results -v Archiving:Backup:Rear rear-2.7 | grep '^15.5'
15.5  x86_64   rear-2.7  succeeded
15.5  ppc64le  rear-2.7  succeeded

# curl https://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/repodata/repomd.xml.key | gpg
...
pub   rsa2048 2020-01-17 [SC] [expired: 2024-05-25]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F
```

**Comment by jsmeix** [comment]
Got that beast!

See in my previous
https://github.com/rear/rear/issues/3289#issuecomment-2242869945
```
# osc results -v Archiving:Backup:Rear rear-2.7 | grep '^15.5'
15.5  x86_64   rear-2.7  outdated (was: finished: unchanged)
15.5  ppc64le  rear-2.7  succeeded
```

That `finished: unchanged` triggered something in my mind:

I think I vaguely remember that OBS does not re-publish
a rebuilt package, when the build result is `unchanged`
(i.e. `osc rebuild` rebuilds but that does not enforce
to also re-publish it).

So I tried what happens when I artificially change
something so that the build result gets `changed`.

I did in rear.spec spelling fixes in comments 'rear' -> 'ReaR'
see
https://build.opensuse.org/package/revisions/Archiving:Backup:Rear/rear-2.7
and the actual changes
https://build.opensuse.org/package/rdiff/Archiving:Backup:Rear/rear-2.7?linkrev=base&rev=4

This automatically triggered a rebuild of
Archiving:Backup:Rear rear-2.7 for all build targets
and now - after that rebuild finished - I get
```
# curl https://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/repodata/repomd.xml.key | gpg
...
pub   rsa2048 2020-01-17 [SC] [expires: 2026-09-30]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F
```
which now matches what
```
osc signkey Archiving:Backup:Rear | gpg
```
shows, cf. my above
https://github.com/rear/rear/issues/3289#issuecomment-2242860385

**Comment by thomasmerz** [comment]
Great! Thanks a lot ❤️ You fixed this issue:
```
$ zypper ref -f 1
Forcing raw metadata refresh
Retrieving repository 'Relax-and-Recover snapshot packages (15.5)' metadata ..................................................................[done]
Forcing building of repository cache
Building repository 'Relax-and-Recover snapshot packages (15.5)' cache .......................................................................[done]
Specified repositories have been refreshed.



$ curl https://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/repodata/repomd.xml.key 2>/dev/null | gpg
gpg: WARNING: no command supplied.  Trying to guess what you mean ...
pub   rsa2048 2020-01-17 [SC] [expires: 2026-09-30]
      985D26DB176472E45FFE5FF51B4DD1E2FFC8DD5F
uid           Archiving:Backup:Rear OBS Project <Archiving:Backup:Rear@build.opensuse.org>
```

**Comment by thomasmerz** [comment]
You fixed it, you got the praise 👏🏼   
My repo was already latest/correct, but the name was something old. Thanks for your hint I had to check with your "URI: https://download.opensuse.org/repositories/Archiving:/Backup:/Rear/15.5/"

**Comment by jsmeix** [comment]
:-)
actually I did not fix it (i.e. I did not correct a bug)
but I learned now how to work around this bug in OBS:

It seems OBS does not re-build or re-publish a package
when metadata changed (at least not when expiration date
of a key changed).

I think when metadata of a published package changed,
then OBS should normally (i.e. except exceptions)
automatcally re-build and re-publish the package.

I think such an enhanced re-build and re-publish behaviour
would be in line with the current automated re-build
of a package which is:
When the package sources changed but also when
another package changed that is needed to build
(i.e. when something changed that gets installed by OBS
into its build system), simply put: when the build system
of a package changed, then the package gets automatically
re-built by OBS.

Bottom line (or "lesson learned"):
I think for all ReaR source packages in OBS
I should do some artificial source change once a year
to trigger a complete re-build and re-publish.

**Comment by jsmeix** [comment]
I did the same artificial source change as in
https://github.com/rear/rear/issues/3289#issuecomment-2242912000
now also for Archiving:Backup:Rear rear-2.6
to trigger a complete re-build and re-publish.

**Comment by jsmeix** [comment]
I plan to remove the old and outdated
ReaR versions 2.5 and 2.4
from Archiving:Backup:Rear see
https://github.com/rear/rear/issues/3290
